src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Listeners/EventHubListener.cs

```diff
@@ -213,24 +229,34 @@ public async Task ProcessEventsAsync(PartitionContext context, IEnumerable<Event
                     message.Dispose();
                 }
 
-                // Checkpoint if we processed any events.
-                // Don't checkpoint if no events. This can reset the sequence counter to 0.
-                // Note: we intentionally checkpoint the batch regardless of function 
-                // success/failure. EventHub doesn't support any sort "poison event" model,
-                // so that is the responsibility of the user's function currently. E.g.
-                // the function should have try/catch handling around all event processing
-                // code, and capture/log/persist failed events, since they won't be retried.
-                if (hasEvents)
+                if (_listenerDisposed.IsDisposed && isAnyInvocationFailed)
                 {
-                    await CheckpointAsync(context);
+                    // Host for the EventHubListener have beed disposed,
+                    // the invocation(s) can be uncompleted in this case

```

Comment:
Rather than saying "uncompleted", could you comment that we expect function invocations to fail before reaching customer code and so the checkpoint should not be advanced?